XenoHolographic-it/IAR-SDK-Sample-iOS
# IAR-4934 Implement ScreenShot and RecordVideo [Target]

# Overview

- Add two buttons (Screen Shot and Record Video) and a progress bar. 
- Implement TakeScreenShot method, StartRecording Method and StopRecordingMethod

# Resources

Jira: https://iversoft.atlassian.net/browse/IAR-4934

# Technical Considerations

- For this sample, it records the IARView for 5 seconds and automatically stops it, showing the share activity modal.
- The Record video buttons is inactive while recording
- The progress view is shown and updated using the progress callback (stopRecording method) while the video is being created.

# Known Issues

None.

# Testing Instructions

**Screenshot:** Open the app, select "Scan an AR Target" and press "SCREENSHOT". It should save the image on IARView and present the share modal.
**Recording:** Open the app, select "Scan an AR Target" and press "RECORD VIDEO". It should disable the button, record the IARView for 5 seconds, show the progress view when preparing the video and at the end, show the share modal. The button should be active again and the progress view shouldn't be visible anymore.

# Screenshots

![IAR4934_screenshot01](https://user-images.githubusercontent.com/96146050/149211056-9328f80c-b9b4-43df-89a1-7afe50e7941f.png)
